pkg/apic/auth/apicauth.go

```diff
@@ -374,6 +376,7 @@ type platformTokenGetter struct {
 	*platformTokenGenerator
 	*keyReader
 	*tokenHolder
+	getTokenMutex *sync.Mutex
```

Comment:
My only issue with the mutex being here is if multiple processes create different token getters.  All we are locking is that the one instantiation is not requesting multiple at a time.  If the lock was a global in the package we can ensure that only one token getter can request a token at a time. Not sure if that matters or not, however.  @vivekschauhan thoughts?

Reply:
I think, locking on the instance should be good. Globally locking will lock each instance which might not be expected as instances can be configured with different configuration.